lib/ansible/modules/system/ufw.py

```diff
@@ -336,14 +353,19 @@ def ufw_version():
             if (ufw_major == 0 and ufw_minor >= 35) or ufw_major > 0:
                 cmd.append([params['comment'], "comment '%s'" % params['comment']])
 
-            execute(cmd)
+            (_, rules_dry, _) = execute(cmd)
+            if module.check_mode:
+                if rules_dry.find("Rules updated") != -1:
```

Comment:
Doesn't work for me when deleting non-existing rules: the output of `--dry-run` will always contain `Rules updated` for me. (It does work for adding rules, though.)

Reply:
That's silly, It works for me :)

I'm using Ubuntu 18.04 LTS and UFW version 0.35. I tested my dev with the "Rules updated" check on this version. I have also checked on Ubuntu 16.04 LTS and it's the same version of UFW. 

What is your OS and UFW version ?

Reply:
To complete here it is the output of commands on my dev computer (Ubuntu 18.04)

#LANG=en_US ufw status verbose
```
To                         Action      From
--                         ------      ----
7000/tcp                   ALLOW IN    Anywhere                  
7000/tcp (v6)              ALLOW IN    Anywhere (v6)             

7000                       ALLOW FWD   Anywhere                  
7000 (v6)                  ALLOW FWD   Anywhere (v6)   
```

# LANG=en_US ufw --dry-run route allow 7000 
```
Skipping adding existing rule
Skipping adding existing rule (v6)
```

Reply:
I'm also using `ufw 0.35`, both with Arch Linux and Debian 9 (stretch).

I also get the `Skipping adding existing rule` message when trying to add a rule which is already there. I don't get such a message when *removing* a rule that doesn't exist. The information `Could not delete non-existent rule` is only printed when run *without* `--dry-run` :(

Reply:
Ok. I understand. In this case, I'll compare content of /etc/user.rules & /etc/user6.rule with the result of `ufw --dry-run delete <my rule>`

Reply:
Yes, that's probably the best way to proceed...